scripts/peer/peer.py

```diff
@@ -0,0 +1,273 @@
+    model.setNk(10)
+    model.update()
+
+    # Calculate and save the PEER factors
+    factors = model.getX()
+    np.savetxt(factors_output_path, factors, delimiter=',')
+    # Calculate and save the weights for each factor
+    weights = model.getW()
+    np.savetxt(weights_output_path, weights, delimiter=',')
+    # Calculate and save the precision values
+    precision = model.getAlpha()
+    np.savetxt(precision_output_path, precision, delimiter=',')
+    # Calculate and save the residuals
+    residuals = model.getResiduals()
+    np.savetxt(residuals_output_path, residuals, delimiter=',')
+
+
+if __name__ == "__main__":
+    import sys
+    run_peer(sys.argv[1], sys.argv[2], sys.argv[3], sys.argv[4], sys.argv[5], sys.argv[6])
+
+EOT"""
+    )
+
+    job.command(
+        f'python run_peer.py {expression_file} {covariates_file} {job.factors_output_path} {job.weights_output_path} {job.precision_output_path} {job.residuals_output_path}'
+    )
+    job.command('ls -l')
+
+    return job
+
+
+# @click.command()
+# @click.option('--expression-file')
+def process_cell_type_on_batch(
+    batch: hb.Batch,
+    cell_type_name: str,
+    expression_file,
+    scores_path=SCORES_PATH,
+    covariates_path=COVARIATES_PATH,
+    sample_id_keys_path=SAMPLE_ID_KEYS_PATH,
+):
+    """
+    Run PEER calculation in hail batch
+    """
+    expression_f = batch.read_input(expression_file)
+    job_prefix = f'{cell_type_name}: '
+    load_data = batch.new_python_job(job_prefix + 'load covariates')
+    intermediate_tuple = load_data.call(
+        get_covariates, scores_path, covariates_path, expression_f, sample_id_keys_path
+    )
+    covariates_csv = load_data.call(get_at_index, intermediate_tuple, 0).as_str()
+    expression_csv = load_data.call(get_at_index, intermediate_tuple, 1).as_str()
+    peer_job = batch.new_job(job_prefix + 'peer')
+    run_peer_job(peer_job, expression_csv, covariates_csv)
+
+    batch.write_output(
+        peer_job.factors_output_path,
+        output_path(f'{cell_type_name}_peer_factors_file.txt'),
+    )
+    batch.write_output(
+        peer_job.factors_output_path, output_path(f'{cell_type_name}_weights_file.txt')
+    )
+    batch.write_output(
+        peer_job.factors_output_path,
+        output_path(f'{cell_type_name}_precision_file.txt'),
+    )
+    batch.write_output(
+        peer_job.factors_output_path,
+        output_path(f'{cell_type_name}_residuals_file.txt'),
+    )
+
+
+def find_cell_types_from_path(path_to_cell_files):
+    """
+    Get cell types from expression data path
+    """
+
+    assert path_to_cell_files.startswith('gs://')
+    cs_client = storage.Client()
+
+    bucket_name = path_to_cell_files.split('gs://')[1].split('/')[0]
+    bucket = cs_client.get_bucket(bucket_name)
+    bucket_path = path_to_cell_files.split(f'gs://{bucket_name}/')[-1]
+
+    logging.info(f'Going to fetch cell types from {bucket_path}')
+    blobs = bucket.list_blobs(prefix=bucket_path)
```

Comment:
This is important as otherwise it gets all files in all subdirectories of the path.

```suggestion
    blobs = bucket.list_blobs(prefix=bucket_path, delimiter='/')
```